includes/class-wc-payments-account.php

```diff
@@ -780,6 +783,22 @@ public function maybe_handle_onboarding() {
 		}
 
 		if ( isset( $_GET['wcpay-connect'] ) && check_admin_referer( 'wcpay-connect' ) ) {
+			$incentive = ! empty( $_GET['promo'] ) ? sanitize_text_field( wp_unslash( $_GET['promo'] ) ) : '';
+
+			// Track connection start with promo incentive.
+			if ( ! isset( $_GET['wcpay-connect-jetpack-success'] )
```

Comment:
Good question. I am split about this. On the one hand I am inclined to track as much as possible, on the other I don't want to track things no one else will use.

I am thinking about adding PO information to this and track it independent of promo. I guess that is on your mind 😁

Reply:
The reason I ask is because @daquinons and I were talking while working on the tracking funnel for PO about how events like this would be very helpful to know when an onboarding starts, independent of where the merchant came from (because we currently rely on events like the user clicking the connect button on the onboarding page, but they could also get there via the task list, or numerous other ways).

It looks like you've basically solved that problem with this PR. The way I see it, these events are very useful for the funnels, and since the promo information is passed in as a prop, we can either attach or not attach it to the event depending on if the user is enrolled in a promo.

WDYT? 🙂 I know it's possibly a bit out of the scope of this issue and I don't want to delay you unnecessarily. 😁

Reply:
OK! I'm convinced 😅 General events it is. Just you wait for it.